drivers/spi/spi_ll_stm32.c

```diff
@@ -679,6 +679,82 @@ static int transceive(const struct device *dev,
 
 #ifdef CONFIG_SPI_STM32_DMA
 
+#ifdef CONFIG_SOC_SERIES_STM32H7X
+#define STM32_SPI_GET_MEM_REGION(node_id)                                       \
```

Comment:
This seems more broadly useful, can we move it to to one of the devicetree related headers where other DT macros live?

Reply:
I'm doubtful about putting it in `include/zephyr/devicetree/dma.h` or create a new file `include/zephyr/devicetree/mem.h`

Reply:
Second option makes good sense to me, its about inspecting pointers to determine if they lie within devicetree defined memory regions?

Reply:
I have taken a look at the rest of the code and the second option seems inconsistent with what Zephyr does. For instance, the driver `gpio.h` declares a `struct gpio_dt_spec` and provides a macro `GPIO_DT_SPEC_GET_BY_IDX` to initialize it. To follow the same pattern, I would put this new struct and macro in the memory driver header. The problem is that there are no memory drivers in Zephyr. The next most similar option seems to me to use `zephyr/include/zephyr/drivers/dma.h`, as this is very related to DMA. That will work for the struct and macro, but what to do with the functions? They are generic, don't belong to a particular driver, soc or board. They only place in which they could fit is somewhere under `subsys`. There is nothing related to memory there yet, so I would need to create a new directory etc. to take this option. 

Another option is, as explained above, to put the struct and the macro in the DMA driver header file and add there the functions as well as `static inline`, header-defined functions. I'm not a big fan of header-defined functions, however, in this case, it might be the only reasonable option.

Which option do you find better?

Reply:
I think I'm going to go with the last option, seems the best one given the circumstances.

Reply:
In the end I have opted for the DMA option as it seemed simpler and I didn't need to modify kernel files.